example/memcached-operator/controller/resource/services.go

```diff
@@ -2,39 +2,179 @@ package resource
 
 import (
 	"context"
+	"fmt"
 
 	examplev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/example/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/example/memcached-operator/controller/key"
 	"github.com/giantswarm/operatorkit/example/memcached-operator/logger"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
 )
 
 const (
 	servicesName = "services"
 )
 
 type ServicesConfig struct {
+	K8sClient kubernetes.Interface
 }
 
 type Services struct {
+	k8sClient kubernetes.Interface
 }
 
 func NewServices(config ServicesConfig) (*Services, error) {
-	s := &Services{}
-	return s, nil
+	d := &Services{
+		k8sClient: config.K8sClient,
+	}
+
+	return d, nil
 }
-func (s *Services) Name() string {
+func (d *Services) Name() string {
```

Comment:
Why `d` though? The resource is `service` and it was `s` which was just right IMO.

Reply:
Yes, thanks I just did bunch of substitutions and missed this detail.